diff --git a/src/app/features/settings/general/General.tsx b/src/app/features/settings/general/General.tsx
index b861a06..a0608ec 100644
--- a/src/app/features/settings/general/General.tsx
+++ b/src/app/features/settings/general/General.tsx
@@ -354,6 +354,49 @@ function Appearance() {
   );
 }
 
+function Gestures() {
+  const [enableBottomGesture, setEnableBottomGesture] = useSetting(settingsAtom, 'enableBottomGesture');
+  const [displayBottomGestureHint, setDisplayBottomGestureHint] = useSetting(settingsAtom, 'displayBottomGestureHint');
+
+  const handleBottomGestureChange = (value: boolean) => {
+    setEnableBottomGesture(value);
+    QML.toggleBottomBar(value);
+    if (value === false) {
+        // if the bottom gesture is turned off, also turn off the gesture hint
+        setDisplayBottomGestureHint(false);
+        QML.toggleDisplayHint(false);
+    }
+  };
+
+  const handleDisplayHintChange = (value: boolean) => {
+    setDisplayBottomGestureHint(value);
+    QML.toggleDisplayHint(value);
+  };
+
+  return (
+    <Box direction="Column" gap="100">
+      <Text size="L400">Gestures</Text>
+      <SequenceCard
+        className={SequenceCardStyle}
+        variant="SurfaceVariant"
+        direction="Column"
+        gap="400"
+      >
+        <SettingTile
+          title="Bottom Navigation Gesture"
+          description="Enable bottom gesture to swipe left or right for navigating backward and forward."
+          after={<Switch variant="Primary" value={enableBottomGesture} onChange={handleBottomGestureChange} />}
+        />
+        <SettingTile
+          title="Bar Indicator"
+          description="Enable visual indicator bar for bottom gesture."
+          after={<Switch variant="Primary" value={displayBottomGestureHint} onChange={handleDisplayHintChange} />}
+        />
+      </SequenceCard>
+    </Box>
+  );
+}
+
 type DateHintProps = {
   hasChanges: boolean;
   handleReset: () => void;
@@ -1003,6 +1046,7 @@ export function General({ requestClose }: GeneralProps) {
           <PageContent>
             <Box direction="Column" gap="700">
               <Appearance />
+              <Gestures />
               <DateAndTime />
               <Editor />
               <Messages />
