diff --git a/src/app/components/Pdf-viewer/PdfViewer.tsx b/src/app/components/Pdf-viewer/PdfViewer.tsx
index 9c7fd98..de0362a 100644
--- a/src/app/components/Pdf-viewer/PdfViewer.tsx
+++ b/src/app/components/Pdf-viewer/PdfViewer.tsx
@@ -79,6 +79,7 @@ export const PdfViewer = as<'div', PdfViewerProps>(
 
     const handleDownload = () => {
       FileSaver.saveAs(src, name);
+      QML.handleDownload(name);
     };
 
     const handleJumpSubmit: FormEventHandler<HTMLFormElement> = (evt) => {
