diff --git a/src/app/features/settings/about/About.tsx b/src/app/features/settings/about/About.tsx
index 5a80e06..76305f0 100644
--- a/src/app/features/settings/about/About.tsx
+++ b/src/app/features/settings/about/About.tsx
@@ -233,6 +233,11 @@ export function About({ requestClose }: AboutProps) {
                         .
                       </Text>
                     </li>
+                    <li>
+                        <Text size="T300">
+                            The <a href="https://github.com/nitanmarcel/cinny-click-packaging" target="_blank" rel="noreferrer noopener">cinny-click-packaging</a> for Ubuntu Touch is © <a href="https://github.com/nitanmarcel" target="_blank" rel="noreferrer noopener">Marcel Alexandru Nitan</a> used under the terms of <a href="https://www.gnu.org/licenses/gpl-3.0.en.html" target="_blank" rel="noreferrer noopener">GNU GENERAL PUBLIC LICENSE v3.0</a>.
+                        </Text>
+                    </li>
                   </Box>
                 </SequenceCard>
               </Box>
