diff --git a/src/app/state/settings.ts b/src/app/state/settings.ts
index 9d97919..1b4b913 100644
--- a/src/app/state/settings.ts
+++ b/src/app/state/settings.ts
@@ -35,6 +35,9 @@ export interface Settings {
   isNotificationSounds: boolean;
 
   developerTools: boolean;
+
+  enableBottomGesture: boolean;
+  displayBottomGestureHint: boolean;
 }
 
 const defaultSettings: Settings = {
@@ -64,6 +67,9 @@ const defaultSettings: Settings = {
   isNotificationSounds: false,
 
   developerTools: false,
+
+  enableBottomGesture: true,
+  displayBottomGestureHint: true,
 };
 
 export const getSettings = () => {
